lib/parse.rb

```diff
@@ -4,38 +4,57 @@ class Parse
   def initialize(path)
     @path = path
     @lines = []
-  end
+    @game_info = {}
 
-  def fopen
-    if File.exists?(@path)
-      file = File.open(@path)
-      begin
-        @lines = file.readlines.map(&:chomp)
-      rescue
-        return nil
-      ensure
-        file.close
+    begin
+      if(File.exist?(@path))
+        @lines = File.readlines(@path, chomp: true)
+        @file = File.basename(@path)
       end
+    rescue Errno::ENOENT
+      puts 'Invalid Path'
     end
   end
 
-  def get_first_line
-    return @lines[0]
+  def obtain_first_line
+    @lines[0]
   end
 
   def output
-    my_object = {
-      "#{@path}": {
-        lines: count_lines()
-      } 
+    @game_info = {
```

Comment:
Não há necessidade de declarar isso como uma variável de instância. Vc pode retornar o resultado direto
```suggestion
    game_info = {
```